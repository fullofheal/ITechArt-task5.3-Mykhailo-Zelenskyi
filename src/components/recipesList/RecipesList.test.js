import { cleanup, screen, render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../store';
import addRecipeActions from '../../store/actions/addRecipeActions';
import RecipesList from './RecipesList';

afterEach(() => {
  cleanup();
});

const blankRecipe = {
  id: '',
  country: '',
  category: '',
  instructions: '',
  name: '',
  tags: '',
  ingredients: [['','']],
  picture: '',
  video: ''
};

describe('test if component is rendered on the page', () => {
  test('should render RecipesList component', () => {
    render(
      <Provider store={store}>
        <RecipesList />
      </Provider>
    );
    const RecipesListComponent = screen.getByTestId('RecipesList');
    expect(RecipesListComponent).toBeInTheDocument();
  });
});

test('should add new recipe and display it in the list', () => {
  const newRecipe = {
    ...blankRecipe,
    name: 'Lasagna',
    id: 123
  };
  store.dispatch(addRecipeActions.addToLocal(newRecipe));
  render(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  );
  expect(screen.getByText('Lasagna')).toBeInTheDocument();
});  

test('should change recipe name in the list', () => {
  const newRecipe = {
    ...blankRecipe,
    id: 123,
    name: 'Lasagna'
  };
  const newRecipeChanged = {
    ...newRecipe,
    id: 123,
    name: 'Pasta'
  }
  store.dispatch(addRecipeActions.addToLocal(newRecipe));
  store.dispatch(addRecipeActions.addToLocal(newRecipeChanged));
  render(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  );
  expect(screen.getByText('Pasta')).toBeInTheDocument();
  expect(screen.queryByText('Lasagna')).not.toBeInTheDocument();
});  

test('should update recipe tags and country of origin', () => {
  const newRecipe = {
    ...blankRecipe,
    id: 123,
    name: 'Lasagna'
  };
  const newRecipeChanged = {
    ...newRecipe,
    id: 123,
    tags: 'dinner',
    country: 'italian'
  }
  store.dispatch(addRecipeActions.addToLocal(newRecipe));
  store.dispatch(addRecipeActions.addToLocal(newRecipeChanged));
  render(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  );
  expect(screen.getByText('Tags: dinner')).toBeInTheDocument();
  expect(screen.getByText('Country of origin: italian')).toBeInTheDocument();
  expect(screen.getByText('Lasagna')).toBeInTheDocument();
});  

test('should remove Pasta recipe from the list', () => {
  const newRecipe1 = {
    ...blankRecipe,
    id: 123,
    name: 'Lasagna'
  };
  const newRecipe2 = {
    ...blankRecipe,
    id: 124,
    name: 'Pasta'
  };
  const newRecipe3 = {
    ...blankRecipe,
    id: 125,
    name: 'Pizza'
  };
  store.dispatch(addRecipeActions.addToLocal(newRecipe1));
  store.dispatch(addRecipeActions.addToLocal(newRecipe2));
  store.dispatch(addRecipeActions.addToLocal(newRecipe3));
  store.dispatch(addRecipeActions.removeFromLocal(124));
  render(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  );
  expect(screen.getByText('Lasagna')).toBeInTheDocument();
  expect(screen.getByText('Pizza')).toBeInTheDocument();
  expect(screen.queryByText('Pasta')).not.toBeInTheDocument();
});  

test('should remove all recipes but still render RecipesList component', () => {
  store.dispatch(addRecipeActions.removeFromLocal(123));
  store.dispatch(addRecipeActions.removeFromLocal(125));
  render(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  );
  const RecipesListComponent = screen.getByTestId('RecipesList');
  expect(RecipesListComponent).toBeInTheDocument();
});

test('RecipesList matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});