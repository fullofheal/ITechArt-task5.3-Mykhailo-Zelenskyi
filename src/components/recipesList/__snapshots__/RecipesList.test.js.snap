// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipesList matches snapshot 1`] = `
<div
  className="recipes__list"
  data-testid="RecipesList"
>
  <h2>
    List of all added recipes
  </h2>
  <input
    className="recipes__search"
    onChange={[Function]}
    placeholder="Search by tags, name, or country of origin"
    type="text"
    value=""
  />
</div>
`;
